src/Build/Graph/GraphBuilder.cs

```diff
@@ -495,24 +496,55 @@ void AddGraphBuildGlobalVariable(PropertyDictionary<ProjectPropertyInstance> glo
             }
         }
 
-        private ParsedProject ParseProject(ConfigurationMetadata configurationMetadata)
+        private ParsedProject ParseProject(ConfigurationMetadata configurationMetadata, bool enableDynamicPlatformResolution)
         {
             // TODO: ProjectInstance just converts the dictionary back to a PropertyDictionary, so find a way to directly provide it.
             var globalProperties = configurationMetadata.GlobalProperties.ToDictionary();
-
-            var projectInstance = _projectInstanceFactory(
-                configurationMetadata.ProjectFullPath,
-                globalProperties,
-                _projectCollection);
+            ProjectGraphNode graphNode;
+            ProjectInstance projectInstance;
+           
+            if (!enableDynamicPlatformResolution)
+            {
+                projectInstance = _projectInstanceFactory(
+                                    configurationMetadata.ProjectFullPath,
+                                    globalProperties,
+                                    _projectCollection);
+            }
+            else
+            {
+                projectInstance = _projectInstanceFactory(
+                                    configurationMetadata.ProjectFullPath,
+                                    null,
+                                    _projectCollection);
+            }
```

Comment:
~~I don't recall this change, why is it needed?~~

It clicked as I was writing that comment. This is the "clean evaluation" to grab data.

```suggestion
                projectInstance = _projectInstanceFactory(
                                    configurationMetadata.ProjectFullPath,
                                    enableDynamicPlatformResolution ? null : globalProperties, // Platform negotiation requires an evaluation with no global properties first
                                    _projectCollection);
```